Class.test.js

```diff
@@ -1,8 +1,211 @@
 require('./MooTools-Core-1.6.0.js');
 
-describe('Class', () => {
+describe('Basics', () => {
    it('creates a new JS class', () => {
       const C = new Class({});
       expect(new C()).toBeInstanceOf(C);
    });
+
+   it('supports being passed a constructor function', () => {
+      const C = new Class(function (name) {
+         this.name = name;
+      });
+      const instance = new C('fish');
+      expect(instance).toBeInstanceOf(C);
+      expect(instance.name).toBe('fish');
+   });
+});
+
+describe('Extends', () => {
+   it('uses Extends to extend classes', () => {
+      const Parent = new Class();
+
+      const C = new Class({
+         Extends: Parent,
+      });
+
+      expect(new C()).toBeInstanceOf(C);
+      expect(new C()).toBeInstanceOf(Parent);
+   });
+
+   /** See tests below; there are many caveats around this */
+   it('can extend native JS classes', () => {
+      class Parent {}
+
+      const C = new Class({
+         Extends: Parent,
+      });
+
+      expect(new C()).toBeInstanceOf(C);
+      expect(new C()).toBeInstanceOf(Parent);
+   });
+
+   it('calls the parent constructor when creating a new Class', () => {
+      const constructor = jest.fn();
+      class Parent {
+         constructor() {
+            constructor();
+         }
+      }
+
+      new Class({
+         Extends: Parent,
+      });
+
+      expect(constructor).toHaveBeenCalled();
+   });
+
+   it('does not call the parent constructor when instantiating a Class', () => {
+      const constructor = jest.fn();
+      class Parent {
+         constructor() {
+            constructor();
+         }
+      }
+
+      const C = new Class({
+         Extends: Parent,
+      });
+
+      new C();
+
+      expect(constructor).toHaveBeenCalledTimes(1);
```

Comment:
I'm a little confused by this case. It `does not call the parent constructor when instantiating a Class`, but this case seems to assert exactly that?